test/datatable.jl

```diff
@@ -109,7 +109,7 @@ module TestDataTable
     dt = DataTable(a=[1, 2], b=[3., 4.])
     dt2 = DataTable(b=["a", "b"], c=[:c, :d])
     @test isequal(merge!(dt, dt2), dt)
-    @test isequal(dt, DataTable(a=[1, 2], b=["a", "b"], c=[:c, :d]))
+    @test isequal(dt, DataTable(a=[1, 2], b=[3., 4.], b_1=["a", "b"], c=[:c, :d]))
```

Comment:
The generic `merge(!)` methods use the value from the last argument in case of conflict, so I think we should respect this. It could make sense to also support passing a `combine` function.

Maybe that's an argument to keep `hcat`, with a different behavior. For that function, the situation is very similar to that faced by `NamedArray` or `AxisArray`: duplicate column names are not allowed, so either an error must be thrown, or names have to be made unique. I would suggest throwing an error by default, with a boolean keyword argument to make names unique when needed; but other solutions can be considered (we could discuss that with the authors of the two named array packages).